pspipe_utils/best_fits.py

```diff
@@ -175,33 +189,37 @@ def get_all_best_fit(spec_name_list, l_th, cmb_dict, fg_dict, spectra, nl_dict=N
     ps_all_th, nl_all_th = {}, {}
 
     for spec_name in spec_name_list:
-        na, nb = spec_name.split("x")
-        if len(na.split("&")) == 2:
-            sv_a, ar_a = na.split("&")
-            sv_b, ar_b = nb.split("&")
+        ms_a, ms_b = spec_name.split("x")
+        if len(ms_a.split(char)) == 2:
+            sv_a, ar_a = ms_a.split(char)
+            sv_b, ar_b = ms_b.split(char)
             noise_key_a = ar_a
             noise_key_b = ar_b
-        elif len(na.split("&")) == 3:
-            sv_a, ar_a, split_a = na.split("&")
-            sv_b, ar_b, split_b = nb.split("&")
+        elif len(ms_a.split("&")) == 3:
+            sv_a, ar_a, split_a = ms_a.split(char)
+            sv_b, ar_b, split_b = ms_b.split(char)
             noise_key_a = f"{ar_a}_{split_a}"
             noise_key_b = f"{ar_b}_{split_b}"
 
         for spec in spectra:
-            ps_all_th[na, nb, spec] = cmb_dict[spec] + fg_dict[f"{sv_a}_{ar_a}", f"{sv_b}_{ar_b}"][spec]
+        
+            ps_all_th[ms_a, ms_b, spec] = cmb_dict[spec] + fg_dict[f"{sv_a}_{ar_a}", f"{sv_b}_{ar_b}"][spec]
+            ps_all_th[ms_b, ms_a, spec] = ps_all_th[ms_a, ms_b, spec].copy()
 
             if bl_dict is not None:
-                ps_all_th[na, nb, spec] *= bl_dict[sv_a, ar_a] * bl_dict[sv_b, ar_b]
-
-            ps_all_th[nb, na, spec] = ps_all_th[na, nb, spec]
+                X, Y = spec
+                ps_all_th[ms_a, ms_b, spec] *=  bl_dict[sv_a, ar_a][X] * bl_dict[sv_b, ar_b][Y]
+                if ms_a != ms_b:
+                    # the if avoid a repetition in the case ms_a == ms_b
+                    ps_all_th[ms_b, ms_a, spec] *= bl_dict[sv_b, ar_b][X] * bl_dict[sv_a, ar_a][Y]
 
             if nl_dict is not None:
                 if sv_a == sv_b:
-                    nl_all_th[na, nb, spec] = nl_dict[sv_a, noise_key_a, noise_key_b][spec]
+                    nl_all_th[ms_a, ms_b, spec] = nl_dict[sv_a, noise_key_a, noise_key_b][spec]
                 else:
-                    nl_all_th[na, nb, spec] = 0
+                    nl_all_th[ms_a, ms_b, spec] = 0
```

Comment:
I guess here we do not need a vector full of zero (something like `np.zeros_like(nl_dict[sv_a, noise_key_a, noise_key_b][spec])`). I also think it doesn't hurt but maybe 0.0 (`float`) is better to avoid possible bugs (it will be added to vector of floats but never knows)